tests/test_atoms_state.py

```diff
@@ -0,0 +1,350 @@
+        Test the Hubbard interactions `U`, `U'`, and `J` for a given set of Slater integrals.
+        """
+        # Adding `slater_integrals` to the `HubbardInteractions` section
+        hubbard_interactions = HubbardInteractions()
+        if slater_integrals is not None:
+            hubbard_interactions.slater_integrals = slater_integrals * ureg('eV')
+
+        # Resolving U, U', and J from class method
+        (
+            u_interaction,
+            u_interorbital_interaction,
+            j_hunds_coupling,
+        ) = hubbard_interactions.resolve_u_interactions(logger)
+
+        if None not in (u_interaction, u_interorbital_interaction, j_hunds_coupling):
+            assert np.isclose(u_interaction.to('eV').magnitude, results[0])
+            assert np.isclose(u_interorbital_interaction.to('eV').magnitude, results[1])
+            assert np.isclose(j_hunds_coupling.to('eV').magnitude, results[2])
+        else:
+            assert (
+                u_interaction,
+                u_interorbital_interaction,
+                j_hunds_coupling,
+            ) == results
+
+    @pytest.mark.parametrize(
+        'u_interaction, j_local_exchange_interaction, u_effective',
+        [
+            (3.0, 1.0, 2.0),
+            (-3.0, 1.0, -4.0),
+            (3.0, None, None),
+            (None, 1.0, None),
+        ],
+    )
+    def test_u_effective(
+        self,
+        u_interaction,
+        j_local_exchange_interaction,
+        u_effective,
+    ):
+        """
+        Test the effective Hubbard interaction `Ueff` for a given set of Hubbard interactions `U` and `J`.
+        """
+        # Adding `u_interaction` and `j_local_exchange_interaction` to the `HubbardInteractions` section
+        hubbard_interactions = HubbardInteractions()
+        self.add_u_j(hubbard_interactions, u_interaction, j_local_exchange_interaction)
+
+        # Resolving Ueff from class method
+        resolved_u_effective = hubbard_interactions.resolve_u_effective(logger)
+        if resolved_u_effective is not None:
+            assert np.isclose(resolved_u_effective.to('eV').magnitude, u_effective)
+        else:
+            assert resolved_u_effective == u_effective
+
+    def test_normalize(self):
+        """
+        Test the normalization of the `HubbardInteractions`. Inputs are defined as the quantities of the `HubbardInteractions` section.
+        """
+        # ? Is this enough for testing? Can we do more?
+        hubbard_interactions = HubbardInteractions()
+        self.add_u_j(hubbard_interactions, 3.0, 2.0)
+        hubbard_interactions.u_interorbital_interaction = 1.0 * ureg('eV')
+        hubbard_interactions.j_hunds_coupling = 2.0 * ureg('eV')
+
+        hubbard_interactions.normalize(None, logger)
+        assert np.isclose(hubbard_interactions.u_effective.to('eV').magnitude, 1.0)
+        assert np.isclose(hubbard_interactions.u_interaction.to('eV').magnitude, 3.0)
+        assert hubbard_interactions.slater_integrals is None
```

Comment:
The only extra step on the normalizer's side is checking the length of `u_matrix`.
Checking logs isn't that easy (maybe via the logger object?), and is kind of redundant here.

I think this can be removed.